cmd/prepare-storage.go

```diff
@@ -150,29 +151,35 @@ func prepForInitXL(firstDisk bool, sErrs []error, diskCount int) InitActions {
 
 	// All disks are unformatted, proceed to formatting disks.
 	if disksUnformatted == diskCount {
-		// Only the first server formats an uninitialized setup, others wait for notification.
-		if firstDisk { // First node always initializes.
+		// Only the first server formats an uninitialized
+		// setup, others wait for notification.
+		if firstDisk {
 			return FormatDisks
 		}
 		return WaitForFormatting
 	}
 
-	// Total disks unformatted are in quorum verify if we have some offline disks.
+	// Total disks unformatted are in quorum verify if we have
+	// some offline disks.
 	if disksUnformatted >= quorum {
-		// Some disks offline and some disks unformatted, wait for all of them to come online.
+		// Some disks offline and some disks unformatted, wait
+		// for all of them to come online.
 		if disksUnformatted+disksFormatted+disksOffline == diskCount {
 			return WaitForAll
 		}
-		// Some disks possibly corrupted and too many unformatted disks.
+		// Some disks possibly corrupted and too many
+		// unformatted disks.
 		return Abort
 	}
 
-	// Already formatted and in quorum, proceed to initialization of object layer.
-	if disksFormatted >= quorum {
+	// Already formatted and having read-quorum, proceed to
+	// initialization of object layer.
+	if disksFormatted >= readQuorum {
 		if disksFormatted+disksOffline == diskCount {
 			return InitObjectLayer
 		}
-		// Some of the formatted disks are possibly corrupted or unformatted, heal them.
+		// Some of the formatted disks are possibly corrupted
+		// or unformatted, heal them.
```

Comment:
If we still return `WaitForHeal` should the caller proceed to heal? under ReadQuorum?